src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -396,27 +396,27 @@ private function buildNode(
             }
         }
 
-        $translations = $versionInfo->languageCodes;
+        $translations = $versionInfo->getLanguageCodes();
         $previewableTranslations = array_filter(
             $translations,
             fn (string $languageCode): bool => $this->isPreviewable($location, $content, $languageCode)
         );
 
         return new Node(
             $depth,
-            $location->id,
-            $location->contentId,
-            $versionInfo->versionNo,
+            $location->getId(),
+            $location->getContentId(),
+            $versionInfo->getVersionNo(),
             $translations,
             $previewableTranslations,
             '', // node name will be provided later by `supplyTranslatedContentName` method
-            $contentType ? $contentType->identifier : '',
-            $contentType ? $contentType->isContainer : true,
-            $location->invisible || $location->hidden,
+            null !== $contentType ? $contentType->getIdentifier() : '',
+            !(null !== $contentType) || $contentType->isContainer(),
```

Comment:
Double negation should be removed.
```suggestion
            null === $contentType || $contentType->isContainer(),
```